### ISSUE TITLE
Bug: `sort-classes`: Properties referenced in objects are not detected as dependencies

### ISSUE THREAD

**Issue by hugop95** [azat-io/eslint-plugin-perfectionist#245]
### Describe the bug

See code example

### Code example

```ts
class Class {
  b = 1
  a = {
    b: this.b
  }
}
```
This code is considered invalid and requires `a` to be before `b` when it shouldn't.

### ESLint version

8.57.0

### ESLint Plugin Perfectionist version

v3.3.0

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Fixed in e7c113d.
Released in [v3.4.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v3.4.0).
